### PR TITLE
move to digital::v2

### PR DESCRIPTION
**Pull request by golledge**
Tested on stm32f302r8 which is not in the example.

### CODE REVIEW

**Comment by JohnDoneth** [comment]
Thanks for the PR! I think we should probably handle the errors instead of ignoring them if possible, even if it is a breaking API change.

**Comment by golledge** [comment]
Ok, I will have a think about how to it. I am new to rust so any direction is welcome.

**Comment by golledge** [comment]
Well I have add some error handling. If you want any changes just say.

File: src/lib.rs [hunk 1]
```diff
@@ -87,17 +89,17 @@ impl<
         d6: D6,
         d7: D7,
         delay: D,
-    ) -> HD44780<D, EightBitBus<RS, EN, D0, D1, D2, D3, D4, D5, D6, D7>> {
+    ) -> core::result::Result<HD44780<D, EightBitBus<RS, EN, D0, D1, D2, D3, D4, D5, D6, D7>>, Error> {
```
Comment (JohnDoneth) [inline src/lib.rs]: Can we change this to `Result<HD44780<D, EightBitBus<RS, EN, D0, D1, D2, D3, D4, D5, D6, D7>>>` by using the new result type you've defined?

File: src/lib.rs [hunk 2]
```diff
@@ -138,17 +140,17 @@ impl<
         d6: D6,
         d7: D7,
         delay: D,
-    ) -> HD44780<D, FourBitBus<RS, EN, D4, D5, D6, D7>> {
+    ) -> core::result::Result<HD44780<D, FourBitBus<RS, EN, D4, D5, D6, D7>>, Error> {
```
Comment (JohnDoneth) [inline src/lib.rs]: Same here, etc.

File: src/lib.rs [hunk 3]
```diff
@@ -28,6 +28,8 @@ pub struct HD44780<D: DelayUs<u16> + DelayMs<u8>, B: DataBus> {
     display_mode: DisplayMode,
 }
 
+pub struct Error;
```
Comment (JohnDoneth) [inline src/lib.rs]: Instead of re-defining this we should re-import it from elsewhere, like the crate root?

File: src/bus/mod.rs [hunk 4]
```diff
@@ -6,8 +6,12 @@ mod fourbit;
 pub use self::eightbit::EightBitBus;
 pub use self::fourbit::FourBitBus;
 
+pub struct Error;
+pub type Result = core::result::Result<(), Error>;
```
Comment (JohnDoneth) [inline src/bus/mod.rs]: Can we move these to an error.rs file and import them here as `crate::Result`?

**Review by JohnDoneth — COMMENTED** [review]
First off, thanks for making additional changes! I just have a couples changes I'd like to see. I'd like to use the new `type Result<T> = ...` everywhere we can as well as defining the Result & Error types in a separate module so it can be re-used.

**Comment by golledge** [comment]
Hope this is what you were after.

**Comment by JohnDoneth** [comment]
Closed in favor of #15. Thanks for your work!

**Pull request closed without merging**